Library/Homebrew/cask/cask_loader.rb

```diff
@@ -297,15 +304,14 @@ def load(config:)
 
           json_cask[:artifacts].each do |artifact|
             # convert generic string replacements into actual ones
-            artifact = cask.loader.from_h_hash_gsubs(artifact, appdir)
+            artifact = cask.loader.from_h_gsubs(artifact, appdir)
             key = artifact.keys.first
             send(key, *artifact[key])
           end
 
           if json_cask[:caveats].present?
             # convert generic string replacements into actual ones
-            json_cask[:caveats] = cask.loader.from_h_string_gsubs(json_cask[:caveats], appdir)
-            caveats json_cask[:caveats]
+            caveats cask.loader.from_h_string_gsubs(json_cask[:caveats], appdir)
```

Comment:
This was changed to allow `json_cask` to be frozen. We could still use an intermediate variable though in a pinch for readability.